packages/apollo-angular/src/types.ts

```diff
@@ -23,44 +19,56 @@ export interface ExtraSubscriptionOptions {
   useZone?: boolean;
 }
 
-export type MutationResult<TData = any> = FetchResult<TData> & {
+export type MutationResult<TData = any> = ApolloLink.Result<TData> & {
   loading?: boolean;
 };
 
 export type Omit<T, K extends keyof T> = Pick<T, Exclude<keyof T, K>>;
 
-export interface WatchQueryOptionsAlone<
+export type WatchQueryOptionsAlone<
   TVariables extends OperationVariables = EmptyObject,
   TData = any,
```

Comment:
FYI there were a couple changes to the types in `@apollo/client` that I'd like to propose we do in this library as well:

- The generic arguments are now consistently `<TData, TVariables>`. Apollo Client 3 had some cases where it was flipped and it was annoying.
- The default `TData` type is now `unknown` instead of `any` to provide better type safety

Thoughts on bringing those changes over here too?